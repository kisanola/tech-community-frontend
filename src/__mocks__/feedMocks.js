export default {
  author: { username: 'name', picture: 'https://image.png' },
  userType: 'organization',
  image: 'https://image.png',
  description: 'description',
  likesCount: 1,
  createdAt: '2019-09-22',
};
