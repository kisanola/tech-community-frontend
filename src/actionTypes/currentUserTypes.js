export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const SET_IS_AUTH = 'SET_IS_AUTH';
