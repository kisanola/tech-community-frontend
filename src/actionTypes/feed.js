export const TOOGLE_LOADING = 'TOOGLE_LOADING';
export const SET_FEED = 'SET_FEED';
