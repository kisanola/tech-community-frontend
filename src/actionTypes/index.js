export const CHANGE_NAME = 'CHANGE_NAME';
export * from './currentUserTypes';
