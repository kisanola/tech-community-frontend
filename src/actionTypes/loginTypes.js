export const NO_ERROR = 'NO_ERROR';
export const LOGIN = 'LOGIN';
export const LOGIN_STARTED = 'LOGIN_STARTED';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGIN_INPUT_CHANGED = 'LOGIN_INPUT_CHANGED';
