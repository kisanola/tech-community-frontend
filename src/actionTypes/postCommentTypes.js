export const POST_COMMENT_STARTED = 'POST_COMMENT_STARTED';
export const POST_COMMENT_SUCCESS = 'POST_COMMENT_SUCCESS';
export const POST_COMMENT_ERROR = 'POST_COMMENT_ERROR';
export const RESTORE_COMMENT_TICK = 'RESTORE_COMMENT_TICK';