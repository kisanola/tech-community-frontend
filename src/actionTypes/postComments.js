export const TOGGLE_LOADING = 'TOGGLE_LOADING';
export const SET_COMMENTS = 'SET_COMMENTS';
