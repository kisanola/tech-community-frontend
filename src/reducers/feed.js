import * as types from '../actionTypes/feed';
import { FEED_LIMIT } from '../constants';

const reducer = (
  state = {
    items: [],
    loading: true,
    limit: 0,
    organizations: [],
  },
  { type, payload },
) => {
  switch (type) {
    case types.TOOGLE_LOADING:
      return {
        ...state,
        loading: payload,
      };
    case types.SET_FEED:
      return {
        ...state,
        items: [...new Set([...state.items, ...payload])],
        limit: state.limit + FEED_LIMIT,
      };
    case types.SET_FEED_ORGANIZATIONS:
      return {
        ...state,
        organizations: payload,
      };
    case types.CLEAR_FEED:
      return {
        ...state,
        items: [],
      };
    default:
      return state;
  }
};

export default reducer;
