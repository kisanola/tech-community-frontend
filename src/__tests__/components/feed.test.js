import React from 'react';
import { shallow } from 'enzyme';
import feedReducer from '../../reducers/feed';
import { SET_FEED, TOOGLE_LOADING } from '../../actionTypes/feed';
import {
  Feed,
  mapStateToProps,
  mapDispatchToProps,
} from '../../components/Feed/Feed';
import feedMocks from '../../__mocks__/feedMocks';

describe('Feed', () => {
  let wrapper;
  const props = {
    onGetFeed: jest.fn(),
    feed: [feedMocks],
    loading: false,
    limit: 2,
  };

  describe('Feed.jsx', () => {
    test('should render Feed.jsx', () => {
      wrapper = shallow(<Feed {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('State', () => {
    let mapped, dispatch;
    beforeEach(() => {
      dispatch = jest.fn();
      mapped = mapDispatchToProps(dispatch);
    });

    test('should return `mapStateToProps`', () => {
      const expectedState = {
        feed: [feedMocks],
        loading: false,
        limit: 2,
      };
      const state = mapStateToProps({
        feed: { items: [feedMocks], loading: false, limit: 2 },
      });
      expect(state).toEqual(expectedState);
    });

    it('Should call onGetFeed', () => {
      mapped.onGetFeed(2, 0);
      expect(dispatch).toHaveBeenCalled();
    });
  });

  describe('Feed reducer', () => {
    test('should test the reducer with type SET_FEED', () => {
      const state = feedReducer(
        { items: [] },
        { type: SET_FEED, payload: [feedMocks] },
      );
      expect(state.items).toEqual([feedMocks]);
    });

    test('should test the reducer with type TOOGLE_LOADING', () => {
      const state = feedReducer({}, { type: TOOGLE_LOADING, payload: true });
      expect(state.loading).toBe(true);
    });
  });
});
