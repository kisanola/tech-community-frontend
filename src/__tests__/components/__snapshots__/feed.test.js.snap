// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed Feed.jsx should render Feed.jsx 1`] = `ShallowWrapper {}`;
